tools/spatialDraw/DrawingManager.js

```diff
@@ -0,0 +1,758 @@
+                size: this.size,
+                color: this.color,
+                drawingId: this.currentLine.obj.drawingId
+            };
+            this.currentLine = null;
+            this.lastPointTime = 0;
+            this.drawingManager.pushEvent(undoEvent);
+        } else {
+            this.currentLine = null;
+            this.lastPointTime = 0;
+        }
+    }
+
+    /**
+     * Creates a drawing from a serialized version.
+     * @param {THREE.Object3D} parent - The parent object to draw in.
+     * @param {Object} drawing - The serialized object defining the object to be drawn.
+     */
+    drawFromSerialized(parent, drawing) {
+        const brushShape = generateBrushShape(drawing.size);
+        const brushMaterial = generateBrushMaterial(drawing.color);
+
+        const threePoints = drawing.points.map(point => new THREE.Vector3(point.x, point.y, point.z));
+        const curve = new THREE.CatmullRomCurve3(threePoints);
+        const extrudeSettings = {
+            extrudePath: curve,
+            steps: threePoints.length - 1
+        };
+        const geometry = new THREE.ExtrudeGeometry(brushShape, extrudeSettings);
+        const mesh = new THREE.Mesh(geometry, brushMaterial);
+        mesh.drawingId = drawing.drawingId;
+        mesh.serialized = drawing;
+        parent.add(mesh);
+    }
+
+    /**
+     * Sets the drawing size.
+     * @param {number} size - The size.
+     */
+    setSize(size) {
+        super.setSize(size);
+        this.brushShape = generateBrushShape(size);
+    }
+
+    /**
+     * Sets the drawing color.
+     * @param {string} color - The color.
+     */
+    setColor(color) {
+        super.setColor(color);
+        this.brushMaterial = generateBrushMaterial(color);
+    }
+};
+
+DrawingManager.Cursor.Offset = class extends DrawingManager.Cursor {
+    /**
+     * Creates an Offset Cursor.
+     */
+    constructor() {
+        super();
+        this.offset = 500; // TODO: Add ability for user to set this by tapping in the scene, like focusing camera lens
+        this.position = new THREE.Vector3(0, 0, 0);
+        this.raycaster = new THREE.Raycaster();
+    }
+
+    /**
+     * Updates the cursor position.
+     * @param {THREE.Scene} scene - The scene to calculate the position in.
+     * @param {THREE.Camera} camera - The camera used for calculating the cursor position.
+     * @param {Object} pointerEvent - The triggering pointer event.
+     */
+    updatePosition(scene, camera, pointerEvent) {
+        const position = {
+            x: (pointerEvent.pageX / window.innerWidth) * 2 - 1,
+            y: - (pointerEvent.pageY / window.innerHeight) * 2 + 1,
+        };
+
+        this.raycaster.setFromCamera(position, camera);
+
+        const ray = this.raycaster.ray;
+
+        this.position = ray.origin.clone().add(ray.direction.clone().multiplyScalar(this.offset)).applyMatrix4(camera.matrixWorld).applyMatrix4(scene.matrixWorld.clone().invert());
```

Comment:
This line would be a bit less daunting if it were wrapped or split up in some way